### PR TITLE
LPD-33047 Allow style books to be applied to a page based on the frontend token definition applied to the page

### PR DESCRIPTION
**Pull request by algobob**
## Re-sent from #790 

Ticket: https://liferay.atlassian.net/browse/LPD-33047

## Motivation
The user only applies a style book if the theme applied to a page is the same as the one applied to Public Pages. We want the user to be able to assign style books based on different token definitions to different pages of the same site.

## Proposed Solution
It filters the style books based on the frontend token definition applied to the page.

### CODE REVIEW

**Comment by algobob** [comment]
ci:test:sf

**Review by thiago-buarqque — APPROVED** [review]
Looks good to me.

@algobob  I'd just merge the breaking changes commits.
1. [LPD-33047 Fetch frontend token definition based on company id, layout id and theme id](https://github.com/liferay-platform-experience/liferay-portal/pull/796/commits/fce1014a5ebfbb724d500ebcd5604d2a1de5a104)
2. [LPD-33047 Remove old theme matching validation](https://github.com/liferay-platform-experience/liferay-portal/pull/796/commits/a76923eaf4e40429420db11ec9e80b6494d24796)

It's good to put the breaking change message together with the breaking code, so it's easier to find the changes in the future.

I did not give much attention to the frontend side, I am mostly handling SF here, since I am out of context.

**Comment by algobob** [comment]
> Looks good to me.
> 
> @algobob I'd just merge the breaking changes commits.
> 
> 1. [LPD-33047 Fetch frontend token definition based on company id, layout id and theme id](https://github.com/liferay-platform-experience/liferay-portal/pull/796/commits/fce1014a5ebfbb724d500ebcd5604d2a1de5a104)
> 2. [LPD-33047 Remove old theme matching validation](https://github.com/liferay-platform-experience/liferay-portal/pull/796/commits/a76923eaf4e40429420db11ec9e80b6494d24796)
> 
> It's good to put the breaking change message together with the breaking code, so it's easier to find the changes in the future.
> 
> I did not give much attention to the frontend side, I am mostly handling SF here, since I am out of context.

cool, it makes sense for me too. Lets just decide a pattern with Evan. I see many ways of using: "baseline", "semantic version" and same commit as you mentioned. What do you think? cc: @ethib137

**Comment by algobob** [comment]
> > cool, it makes sense for me too. Lets just decide a pattern with Evan. I see many ways of using: "baseline", "semantic version" and same commit as you mentioned. What do you think? cc: @ethib137
> 
> I don't think I understand exactly what the question is. Could you explain a bit more @algobob

Sure. So, we need to add that "breaking change" message template when we're implementing changes that affect `*-api` modules interface. i.e: update params, add new methods...  Some people just add the message in a commit with the new versions like this [one](https://github.com/liferay/liferay-portal/commit/44648ff0b15fe5936f835b5ae5d7570903dab3a3) and other people add the message into the commit with the java code changes, like this [one](https://github.com/liferay/liferay-portal/commit/49b6c1496e13180a173a99a3c9e29b62462e02f8)

Then my question is about which way should we follow?

**Comment by ethib137** [comment]
> > > cool, it makes sense for me too. Lets just decide a pattern with Evan. I see many ways of using: "baseline", "semantic version" and same commit as you mentioned. What do you think? cc: @ethib137
> > 
> > 
> > I don't think I understand exactly what the question is. Could you explain a bit more @algobob
> 
> Sure. So, we need to add that "breaking change" message template when we're implementing changes that affect `*-api` modules interface. i.e: update params, add new methods... Some people just add the message in a commit with the new versions only like this [one](https://github.com/liferay/liferay-portal/commit/44648ff0b15fe5936f835b5ae5d7570903dab3a3) and other people add the message into the commit with the java code changes, like this [one](https://github.com/liferay/liferay-portal/commit/49b6c1496e13180a173a99a3c9e29b62462e02f8)
> 
> Then my question is about which way should we follow?

According to our [documentation](https://liferay.atlassian.net/wiki/spaces/ENGR/pages/2489483265/Process+to+register+breaking+changes+in+commits) on this, the breaking change commit should be the commit where the breaking change occurs in the java code.

**Comment by algobob** [comment]
ci:test:sf

**Comment by algobob** [comment]
> According to our [documentation](https://liferay.atlassian.net/wiki/spaces/ENGR/pages/2489483265/Process+to+register+breaking+changes+in+commits) on this, the breaking change commit should be the commit where the breaking change occurs in the java code.

Cool. Done :heavy_check_mark:

**Comment by ealonso** [comment]
Hey @algobob,

A couple of things since this pr should be reviewed by @liferay-page-management team.

1. You need to add the buildLang commit
2. If the new language key is a sentence it should ends in a "."
3. Please execute source format task after every commit, if not it is really difficult to review your changes, It is not make sense to split your code in commits if you mixed logic between commits and you don't have formatted every commit properly.
4. Please do not create unnecessary components, StyleBookManager, shouldn't be a component, also as I can see it is only used in ContentPageEditorDisplayContext, so it is better if you move all the logic to this class.

Regards,
Eudaldo.

**Comment by algobob** [comment]
> Hey @algobob,
> 
> A couple of things since this pr should be reviewed by @liferay-page-management team.
> 
> 1. You need to add the buildLang commit
> 2. If the new language key is a sentence it should ends in a "."
> 3. Please execute source format task after every commit, if not it is really difficult to review your changes, It is not make sense to split your code in commits if you mixed logic between commits and you don't have formatted every commit properly.
> 4. Please do not create unnecessary components, StyleBookManager, shouldn't be a component, also as I can see it is only used in ContentPageEditorDisplayContext, so it is better if you move all the logic to this class.
> 
> Regards, Eudaldo.

@ealonso 

1.  It seems its not a practice in our team anymore. Also, Brian run it before merging
2.  good catch. updating...
3.  cool. updating...
4.  oh, my idea was to make easier to unit test this new behavior by isolating, instead of make `ContentPageEditorDisplayContext`  even bigger than already is and test it via integration test only.  The `manager` package initially seems to be a good place. Some methods of these managers also are used in one place only i.e: `ContentManager#getFragmentEntryLinkMappedLayoutDisplayPageObjectProviders `, `FormItemManager#addFormStepLayoutStructureItems`, `FragmentCollectionManager#updateSortedFragmentCollectionKeys` ...

  Should I move it to `ContentPageEditorDisplayContext` anyway?

**Comment by algobob** [comment]
ci:test:sf

**Comment by algobob** [comment]
ci:test:sf

**Comment by algobob** [comment]
ci:test:sf

File: modules/apps/layout/layout-content-page-editor-web/src/test/java/com/liferay/layout/content/page/editor/web/internal/display/context/ContentPageEditorDisplayContextTest.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+		Assert.assertEquals(result.toString(), 1, result.size());
+
+		Assert.assertEquals(
+			name,
+			result.get(
+				0
+			).get(
+				"name"
+			));
```
Comment (thiago-buarqque) [inline modules/apps/layout/layout-content-page-editor-web/src/test/java/com/liferay/layout/content/page/editor/web/internal/display/context/ContentPageEditorDisplayContextTest.java:156]: Create a variable for 

```java
result.get(
    0
)
```

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 2]
```diff
@@ -12803,6 +12803,7 @@ only-one-sku-can-be-approved=Only one SKU can be approved.
 only-one-value-is-allowed-in-single-selection-mode=Only one value is allowed in "Single" selection mode.
 only-required-in-this-asset-library=Only Required in This Asset Library
 only-show-results-for-web-content-listed-in-a-web-content-display-widget=Only show results for web content listed in a web content display widget.
+only-style-books-based-on-the-frontend-token-definition-provided-by-x-theme-are-visible=Only style books based on the frontend token definition provided by {0} theme are visible
```
Comment (thiago-buarqque) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:12806]: We probably have a period at the end of the sentence in the design. You should add it here as well

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/page_design_options/components/PageDesignOptionsSidebar.js
Comment (thiago-buarqque) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/page_design_options/components/PageDesignOptionsSidebar.js:1]: Frontend side looks fine to me, but better to have a stronger opinion from @pat270

File: modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/display/context/ContentPageEditorDisplayContext.java [hunk 3]
```diff
@@ -664,6 +665,12 @@ public Map<String, Object> getEditorContext() throws Exception {
 			).put(
 				"styleBookEnabled",
 				() -> {
+					if (FeatureFlagManagerUtil.isEnabled(
+							themeDisplay.getCompanyId(), "LPD-30204")) {
+
+						return true;
+					}
+
 					Layout layout = themeDisplay.getLayout();
```
Comment (thiago-buarqque) [inline modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/display/context/ContentPageEditorDisplayContext.java:674]: Other methods in this that fill the attributes in this `HashMapBuilder` use `Layout`, `Theme` and other info from the themeDisplay to build their values.

**Review by thiago-buarqque — CHANGES_REQUESTED** [review]
Thanks @algobob, just a few comments, let me know if you disagree with any of them

**Comment by algobob** [comment]
ci:test:sf

**Review by thiago-buarqque — APPROVED** [review]
LGTM. @algobob  please move forward with this one asking for the other team's review

**Comment by algobob** [comment]
> ### ❌ ci:test:sf - 0 out of 1 jobs passed in 6 minutes
> Click here for more details.
> #### Base Branch:
> Branch Name: [master](https://github.com/liferay/liferay-portal/tree/master)Branch GIT ID: [7500a1d19d00706e027f3a924643af94b4caa2c5](https://github.com/liferay/liferay-portal/commit/7500a1d19d00706e027f3a924643af94b4caa2c5)
> 
> #### Sender Branch:
> Branch Name: [LPD-33047-2](https://github.com/algobob/liferay-portal/tree/LPD-33047-2)Branch GIT ID: [344857e9ffa9b900ce3b34ba86b5e2b2c1bd46b1](https://github.com/algobob/liferay-portal/commit/344857e9ffa9b900ce3b34ba86b5e2b2c1bd46b1)
> 
> 0 out of 1jobs PASSED
> #### 1 Failed Jobs:
> * [test-portal-source-format](https://test-1-39.liferay.com/job/test-portal-source-format/7146/)
> 
> ##### For more details click [here](https://test-1-39.liferay.com/userContent/jobs/test-portal-source-format/builds/7146/jenkins-report.html).
> format-source-files:
>      [java] java.lang.Exception: Found 1 formatting issues:
>      [java] 1: Do not use org.jsoup, see LPD-42623: ./modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/display/context/ContentPageEditorDisplayContext.java (SourceCheck:IllegalImportsCheck)
>      [java] 
>      [java]   at com.liferay.source.formatter.SourceFormatter.format(SourceFormatter.java:472)
>      [java]   at com.liferay.source.formatter.SourceFormatter.main(SourceFormatter.java:301)
> [stopwatch] [run.batch.test.action: 1:54.851 sec]
>      [echo] The following error occurred while executing this line:
>      [echo] /opt/dev/projects/github/liferay-portal/portal-impl/build.xml:497: The following error occurred while executing this line:
>      [echo] /opt/dev/projects/github/liferay-portal/portal-impl/build.xml:650: Java returned: 1
>   [typedef] Could not load definitions from resource org/apache/maven/artifact/ant/antlib.xml. It could not be found.
>   [taskdef] Could not load definitions from resource org/jacoco/ant/antlib.xml. It could not be found.
>    [delete] Deleting: /opt/dev/projects/github/liferay-portal/null1389395935.properties
> merge-test-results:
> [mkdir] Created dir: /opt/dev/projects/github/liferay-portal/test-results
> [beanshell] Truncating errors in /opt/dev/projects/github/liferay-portal/portal-impl/test-results/TEST-JenkinsLogAssertTest.xml
> [junitreport] Processing /opt/dev/projects/github/liferay-portal/test-results/TESTS-TestSuites.xml to /tmp/null820285745
> [junitreport] Loading stylesheet ile:/opt/java/ant/lib/ant-junit.jar!/org/apache/tools/ant/taskdefs/optional/junit/xsl/junit-frames.xsl
> [junitreport] Transform time: 143ms
> [junitreport] Deleting: /tmp/null820285745
> [echo] A report with all the test results can be found at test-results/html/index.html.
> [mkdir] Created dir: /opt/dev/projects/github/liferay-jenkins-ee/test-results
> [copy] Copying 1 file to /opt/dev/projects/github/liferay-jenkins-ee/test-results
> [echo] run.batch.test.tear.down.start.timestamp: 12-11-2024 09:03:36:073 PST
> [stopwatch] [run.batch.test.tear.down: 0.001 sec]
> [typedef] Could not load definitions from resource org/apache/maven/artifact/ant/antlib.xml. It could not be found.
> [taskdef] Could not load definitions from resource org/jacoco/ant/antlib.xml. It could not be found.
> [delete] Deleting: /opt/dev/projects/github/liferay-portal/null1389395935.properties
> 
> 
> print-gc-logs:
> [echo]

SF failing seems no be related with my changes. Ignoring it...

**Comment by algobob** [comment]
ci:test:relevant

**Comment by algobob** [comment]
ci:test:platform-experience

**Comment by algobob** [comment]
> ### ❌ ci:test:platform-experience - 49 out of 54 jobs passed in 1 hour 5 minutes
> Click here for more details.
> ##### Test bundle downloads:
> * [liferay-portal-bundle-tomcat.tar.gz](https://test-1-37.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/19223/bundles/liferay-portal-bundle-tomcat.tar.gz)

It will be fixed in a different PR . Ignoring it for now...

**Pull request closed without merging**